### PR TITLE
UI: Use SelectableValue as Segment value

### LINKED ISSUE #20483: Segment component displays Value rather than Label
https://github.com/grafana/grafana/blob/ca3dff99e81cecbfa1332112c6c16309788d17dd/packages/grafana-ui/src/components/Segment/Segment.tsx#L21

The label being rendered is the underlying value of the select, not the label that appears in the dropdown itself.

Screen Shot 2019-11-19 at 10 50 22 AM
Screen Shot 2019-11-19 at 10 50 25 AM

### PR DESCRIPTION
**Pull request by sunker**
At the moment, the option value is being displayed in the label. This PR makes sure the label is being displayed correctly. 

fixes #20483

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
LGTM

**Comment by seanlaff** [comment]
Thanks!

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx [hunk 1]
```diff
@@ -112,30 +112,30 @@ export class QueryEditor extends PureComponent<Props, State> {
       <>
         <QueryInlineField label="Region">
           <Segment
-            value={query.region || 'Select region'}
+            value={this.toOption(query.region || 'Select region')}
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: this makes it a bit harder to use the Segment as you have to do this everywhere, maybe supporting `T | SelectValue<T>`  will make it easier

File: packages/grafana-ui/src/components/Segment/Segment.tsx [hunk 2]
```diff
@@ -18,18 +18,23 @@ export function Segment<T>({
   const [Label, width, expanded, setExpanded] = useExpandableLabel(false);
 
   if (!expanded) {
-    return <Label Component={Component || <a className={cx('gf-form-label', 'query-part', className)}>{value}</a>} />;
+    return (
+      <Label
+        Component={Component || <a className={cx('gf-form-label', 'query-part', className)}>{value && value.label}</a>}
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Segment/Segment.tsx]: maybe label could have been looked up from the options

File: public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx (hunk 1 above)
Comment (sunker) [reply public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx]: I agree! Will fix

File: packages/grafana-ui/src/components/Segment/Segment.tsx (hunk 2 above)
Comment (sunker) [reply packages/grafana-ui/src/components/Segment/Segment.tsx]: Wouldn't work for SegmentAsync since it lazyloads options

**Pull request merged**